lib/guest_installation_and_configuration_base.pm

```diff
@@ -2572,12 +2572,9 @@ sub start_guest_installation {
     my $_guest_installation_log = "$common_log_folder/$self->{guest_name}/$self->{guest_name}" . "_installation_log_" . $_start_installation_timestamp;
     assert_script_run("touch $_guest_installation_log && chmod 777 $_guest_installation_log");
     # Dry run always timeout when downloading initrd from download.opensuse.org in O3
-    my $ret = script_run("set -o pipefail; $self->{virt_install_command_line_dryrun} 2>&1 | tee -a $_guest_installation_dryrun_log", timeout => 600 / get_var('TIMEOUT_SCALE', 1), die_on_timeout => 0);
+    my $ret = script_run("set -o pipefail; timeout 580 $self->{virt_install_command_line_dryrun} 2>&1 | tee -a $_guest_installation_dryrun_log", timeout => 600 / get_var('TIMEOUT_SCALE', 1));
     save_screenshot;
-    unless (defined(script_run('set +o pipefail', die_on_timeout => 0))) {
-        reconnect_when_ssh_console_broken;
-        script_run("set +o pipefail");
-    }
+    script_run 'set +o pipefail';
```

Comment:
This does not seem equivalent replacement. reconnect_when_ssh_console_broken is what needed here when the connection is detected to be broken via script_run problem. Suggest to use below:

```
    unless (defined(script_run('timeout 28 set +o pipefail', timeout => 30)) {
        reconnect_when_ssh_console_broken;
        script_run("set +o pipefail");
    }
```

Reply:
`set` is a shell command so we can't call `timeout` on it but we can use

```suggestion
    enter_cmd "set +o pipefail; echo DONE > /dev/$serialdev";
    unless (defined(wait_serial('DONE', timeout => 30)) {
        reconnect_when_ssh_console_broken;
        script_run("set +o pipefail");
    }
```